### PR TITLE
Add select component

### PR DESCRIPTION
**Pull request by cidermitaina**
## What
- Add select component

## Screenshots
image

### CODE REVIEW

**Comment by cidermitaina** [comment]
@nabeliwo @im36-123

Please review !! 🙋‍♀️

File: src/components/Select/Select.tsx [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+        <Base
+          className={classNames}
+          value={value}
+          name={name}
+          required={required}
+          disabled={disabled}
+          onChange={this.handleChange}
+          theme={theme}
+        >
+          {blankOption && <option value=""> {placeholder}</option>}
```
Comment (nabeliwo) [inline src/components/Select/Select.tsx]: [ask] Is there a pattern where blankOption is true and there is no placeholder? 
How about putting out elements by looking at placeholder values?

File: src/components/Select/Select.tsx [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+  currentTarget: {
+    value: string
+  }
+}
+
+class SelectComponent extends React.PureComponent<Props & InjectedProps> {
+  public render() {
+    const {
+      className = '',
+      value = '',
```
Comment (nabeliwo) [inline src/components/Select/Select.tsx]: [MUST] `value` is required, so there is no need to enter a default value.

File: src/components/Select/Select.tsx [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+
+      &::before {
+        border-top: 5px solid ${palette.TEXT_BLACK};
+        top: 52%;
+      }
+
+      &::after {
+        bottom: 52%;
+        border-bottom: 5px solid ${palette.TEXT_BLACK};
+      }
```
Comment (nabeliwo) [inline src/components/Select/Select.tsx]: [IMO] Implement icons with Icon Component for visual unification. There is an icon named 'fa-sort'.

File: src/components/Select/Select.tsx [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+      -webkit-appearance: none;
+      -moz-appearance: none;
+      padding-right: 20px;
+
+      &::placeholder {
+        color: ${palette.TEXT_GREY};
+      }
+
+      &:focus {
+        border-color: ${palette.hoverColor(palette.MAIN)};
```
Comment (nabeliwo) [inline src/components/Select/Select.tsx]: [MUST] I think the color here is palette.MAIN.

File: src/components/Select/Select.tsx [hunk 5]
```diff
@@ -0,0 +1,160 @@
...
+      font-size: ${size.pxToRem(size.font.TALL)};
+      color: ${palette.TEXT_BLACK};
+      line-height: 1.6;
+      outline: none;
+      box-sizing: border-box;
+      appearance: none;
+      -webkit-appearance: none;
+      -moz-appearance: none;
+      padding-right: 20px;
+
```
Comment (nabeliwo) [inline src/components/Select/Select.tsx]: [MUST] Add 'cursor: pointer'.

File: src/components/Select/Select.tsx [hunk 6]
```diff
@@ -0,0 +1,160 @@
...
+        border-color: ${palette.hoverColor(palette.MAIN)};
+      }
+
+      &.error {
+        border-color: ${palette.DANGER};
+      }
+
+      &[disabled] {
+        border-color: ${palette.BORDER};
+        pointer-events: none;
```
Comment (nabeliwo) [inline src/components/Select/Select.tsx]: [MUST] Add 'cursor: not-allowed'.

File: src/components/Select/Select.tsx [hunk 7]
```diff
@@ -0,0 +1,160 @@
+import * as React from 'react'
+import styled, { css } from 'styled-components'
+
+import { InjectedProps, withTheme } from '../../hocs/withTheme'
+
+export interface Options {
```
Comment (nabeliwo) [inline src/components/Select/Select.tsx]: [MUST] Please don't export if you don't use it elsewhere.

File: src/components/Select/Select.tsx [hunk 8]
```diff
@@ -0,0 +1,160 @@
...
+
+export interface Props {
+  className?: string
+  value: string
+  name: string
+  required?: boolean
+  disabled?: boolean
+  error?: boolean
+  width?: number | string
+  options: Options[]
```
Comment (nabeliwo) [inline src/components/Select/Select.tsx]: [MUST] Type name is correct with 'Option'.

File: src/components/Select/Select.tsx [hunk 9]
```diff
@@ -0,0 +1,160 @@
...
+  blankOption?: boolean
+  placeholder?: string
+  onChange?: (name: string, value: string) => void
```
Comment (ouji-miyahara) [inline src/components/Select/Select.tsx]: Please add `aria-label` property for a11y.
And check Accessibility panel in storybook !

image

File: src/components/Select/Select.tsx (hunk 1 above)
Comment (im36-123) [reply src/components/Select/Select.tsx]: I think `blackOption` and `placeholder ` are unnecessary for this component.
The user can do that same by simply passing `{label: 'Fruit', value: '' }`

Comment (cidermitaina) [reply src/components/Select/Select.tsx]: > The user can do that same by simply passing {label: 'Fruit', value: '' }

That's true.
BlackOption and placeholder are unnecessary.

Fixed here.
https://github.com/kufu/smarthr-ui/pull/356/commits/6d7bc5dca026fb0163f65bec0dd9675d83b3e780

File: src/components/Select/Select.tsx (hunk 9 above)
Comment (cidermitaina) [reply src/components/Select/Select.tsx]: Fixed here.
8a7a0ab

I'll be careful to check the accessible panel of the storybook!

**Comment by cidermitaina** [comment]
@nabeliwo @im36-123 @ouji-miyahara 

Thank you for review😺
I fixed for your comments.
Please review again🙇‍♀️


83738ae, ca95f47, 6d7bc5d, 8a7a0ab

image

File: src/components/Select/Select.tsx [hunk 10]
```diff
@@ -0,0 +1,163 @@
...
+      border: ${frame.border.default};
+      background-color: #fff;
+      font-size: ${size.pxToRem(size.font.TALL)};
+      color: ${palette.TEXT_BLACK};
+      line-height: 1.6;
+      outline: none;
+      box-sizing: border-box;
+      appearance: none;
+      -webkit-appearance: none;
+      -moz-appearance: none;
```
Comment (nabeliwo) [inline src/components/Select/Select.tsx]: [MUST] styled-components automatically inserts a`Vendor Prefix`, so there is no need to specify it.

**Comment by cidermitaina** [comment]
@nabeliwo 
Thank you for review many times🙇‍♀️
Please review again🙋‍♀️
f26c0ed

**Pull request merged**